exercises/hexadecimal/hexadecimal_test.go

```diff
@@ -42,18 +43,33 @@ var testCases = []struct {
 
 func TestParseHex(t *testing.T) {
 	for _, test := range testCases {
-		switch out, err := ParseHex(test.in); {
-		case err != nil:
-			if test.errCase == "none" {
+		out, err := ParseHex(test.in)
+		if test.errCase != "none" {
+			// check if err is of error type
+			var _ error = err
+
+			// we expect error
+			if err == nil {
+				t.Errorf("ParseHex(%q): expected an error, but error is nil",
+					test.in)
+			}
+
+			if !strings.Contains(strings.ToLower(err.Error()), test.errCase) {
+				t.Errorf(
+					"ParseHex(%q) returned error %q. Expected error %s.",
```

Comment:
can I ask that this be clarify to say something like `Expected error containing '%s'` so a student who sees this error message can better understand what is required of them?

Sure, they can see it by reading the source, but if it can be made from the error message let's do that. For example it might be difficult to understand what "Expected error range" means.